src/fonduer/learning/dataset.py

```diff
@@ -99,29 +99,31 @@ def _map_to_id(self) -> None:
                 ]
                 s = mark_sentence(mention_to_tokens(candidate[i]), args)
                 self.X_dict[f"m{i}"].append(
-                    torch.LongTensor(
+                    torch.tensor(
                         [
                             self.word2id[w]
                             if w in self.word2id
                             else self.word2id["<unk>"]
                             for w in s
-                        ]
+                        ],
+                        dtype=torch.long,
                     )
                 )
 
     def _map_features(self) -> None:
         self.X_dict.update({"feature_index": [], "feature_weight": []})
         for i in range(len(self.candidates)):
             self.X_dict["feature_index"].append(
-                torch.LongTensor(
+                torch.tensor(
                     self.features.indices[
                         self.features.indptr[i] : self.features.indptr[i + 1]
-                    ]
+                    ],
+                    dtype=torch.long,
                 )
                 + 1
             )
             self.X_dict["feature_weight"].append(
-                torch.Tensor(
+                torch.tensor(
```

Comment:
This change broke the test.

Reply:
The error is RuntimeError: expected backend CPU and dtype Double but got backend CPU and dtype Float and it happened at `return (w.unsqueeze(2) * self.weight(x)).sum(dim=1)` (https://github.com/SenWu/emmental/blob/v0.0.4/src/emmental/modules/sparse_linear_module.py#L66).

Reply:
`self.weight` is `nn.Embedding` and `self.weight(x)` is a tensor of dtype==float32.

Reply:
When an error occurred, `w.unsqueeze(2)` was a tensor of dtype==double.

Reply:
Multiplying two tensor of different dtype (float and double) caused the RuntimeError.

Reply:
torch.Tensor is an alias for the default tensor type (torch.FloatTensor); thus, float32.
torch.tensor() constructor infers dtype from data. Seems like dtype==double was inferred.
Should have explicitly specified dtype.